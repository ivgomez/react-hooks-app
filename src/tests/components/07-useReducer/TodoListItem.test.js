import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import { TodoListItem } from "../../../components/07-useReducer/TodoListItem";

describe("Test on <TodoListItem />", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={0}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );
  test("should rendered ok", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should call the handleDelete Method", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(1);
  });
  test("should call the handleToggle Method", () => {
    wrapper.find("p").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("should display the text ok", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(`1. ${demoTodos[0].desc}`);
  });
  test("should have the complete class if it's done", () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(<TodoListItem todo={demoTodos[0]} />);
    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
