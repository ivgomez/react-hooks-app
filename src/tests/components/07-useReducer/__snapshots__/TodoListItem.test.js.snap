// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoListItem /> should rendered ok 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Learning react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
